near-sdk/src/promise.rs

```diff
@@ -325,21 +325,71 @@ impl BorshSchema for Promise {
 enum PromiseSubtype {
     Single(Rc<PromiseSingle>),
     Joint(Rc<PromiseJoint>),
+    Yielded(PromiseIndex),
 }
```

Comment:
This way, you can't add other actions in the receipt that starts with yielded promise. As a solution, I'd suggest making it a part of `PromiseSingle`:
```rust
enum PromiseSingleSubtype {
    Regular {
        account_id: AccountId,
        after: RefCell<Option<Rc<Promise>>>,
        /// Promise index that is computed only once.
        promise_index: RefCell<Option<PromiseIndex>>,
    },
    Yielded(PromiseIndex),
}

struct PromiseSingle {
    pub subtype: PromiseSingleSubtype,
    pub actions: RefCell<Vec<PromiseAction>>, 
}
```
This way, you can also panic in runtime for `regular_promise.then(yielded_promise)`